pkg/storage/read_as_of_iterator.go

```diff
@@ -129,43 +133,40 @@ func (f *ReadAsOfIterator) updateValid() bool {
 // advance moves past keys with timestamps later than f.asOf and skips MVCC keys
 // whose latest value (subject to f.asOF) has been deleted by a point or range
 // tombstone.
```

Comment:
conceptually: we need to pass `seeked` ==false to get info on `f.newestRangeTombstone`, since `RangeKeyChanged()` may be false, yet we still might have a rangekey to worry about, right?

Next time I make edits to this code, I might add a comment that explains this, or push this logic outside of `advance` and into `readAsOfIterator.SeekGE` to remove the parameter in `advance`.

Reply:
Yes. Because we're getting an external iterator, whose state is unknown to us, the `SeekGE` call may land on a range key that the iterator was already positioned on, so we need to explicitly reset the state on any `SeekGE` call too.